### PR TITLE
Add more characters with lua

### PR DESCRIPTION
**Pull request by tposejank**
Hello again, I dont know what pr this one is, but i sure hope it is interesting.
# What this do lmfao

Create another character in the screen, so you don't have to remove GF for a song like Final Destination.

https://user-images.githubusercontent.com/81495861/147373426-cd91a115-5233-4bfc-892a-d62e63333890.mp4
*Jghost added as a character*

# A script that could give use to this function:
image

You can also remove the character if you wish to, and to change them do that instead but add another one instantly!

`setProperty` for these could be possible, I don't know how though.

Compatible characters:
* GF
* BF (player)
* Dad (opponent)

### CODE REVIEW

File: source/FunkinLua.hx [hunk 1]
```diff
@@ -677,6 +677,36 @@ class FunkinLua {
...
+			var charType:Int = 0;
+			switch (type.toLowerCase()) {
+				case 'boyfriend' | 'bf' | 'player':
+					charType = 0;
+				case 'dad' | 'opponent' | '0':
+					charType = 1;
+				case 'gf' | 'girlfriend' | '1':
+					charType = 2;
+				default:
+					charType = 0;
```
Comment (Steviegt6) [inline source/FunkinLua.hx:690]: You don't need a case for `boyfriend`, `bf`, or `player`.

File: source/FunkinLua.hx [hunk 2]
```diff
@@ -677,6 +677,36 @@ class FunkinLua {
...
```
Comment (Steviegt6) [inline source/FunkinLua.hx:705]: You don't need a case for `boyfriend`, `bf`, or `player`.

**Comment by DaBanana64** [comment]
**I really wanted this since #1532**

**Comment by Doritojoey** [comment]
If I had 20 bucks I most likely would not give it to you but this is FUCKING MAGNIFICENT

**Comment by DMMaster636** [comment]
Thank you. Just thank you. I added this to my fork, bcs it is SO useful.

**Comment by sutartgaming** [comment]
how to make it sing sorry if im look like a dumbass

**Comment by DMMaster636** [comment]
im having problems with making another bf char sing. basically, he doesn't wanna sing at all.
image

**Comment by ethannn123** [comment]
this is GODLY

**Comment by Doritojoey** [comment]
> im having problems with making another bf char sing. basically, he doesn't wanna sing at all. (here is the code) image

Data doesn't continue after 4, you just go back to 1

**Comment by DMMaster636** [comment]
> 

oh, ok, thanks

**Comment by bbpanzu** [comment]
might pull

**Comment by tposejank** [comment]
sdfjksdfjklsdjkljkljkkl

**Comment by monkeystink** [comment]
it doesn't want to compile

**Comment by DMMaster636** [comment]
> it doesn't want to compile

It works tho, you just have to add it manually.

**Comment by monkeystink** [comment]
> > it doesn't want to compile
> 
> It works tho, you just have to add it manually.

well how do i add it manually

**Comment by DMMaster636** [comment]
> > > it doesn't want to compile
> > 
> > 
> > It works tho, you just have to add it manually.
> 
> well how do i add it manually

look at the added code and copy it over to your build

**Comment by Itz-Miles** [comment]
Does this PR work with the most recent release or commit on Psych Engine?

**Comment by DMMaster636** [comment]
> Does this PR work with the most recent release or commit on Psych Engine?

yes. it just needs to be updated so you don't have to manually copy the code

**Comment by ProgramerGamerZone** [comment]
> might pull

YES PLEASE

**Comment by monkeystink** [comment]
the add character doesn't work also i did it like this:

function onCreate()
    addCharacter('pico', 'dad')
end


function opponentNoteHit(id, data, notetype, sus)
    if notetype == 'No Animation' then
        if data == 0 then
            characterPlayAnim('pico', 'singLEFT')
        elseif data == 1 then
            characterPlayAnim('pico', 'singDOWN')
        elseif data == 2 then
            characterPlayAnim('pico', 'singUP')
        elseif data == 3 then
            characterPlayAnim('pico', 'singRIGHT')
        end
    end
end

**Comment by monkeystink** [comment]
tried adding the code now but shit doesn't work, like i don't even know where should i put the code and what to do with it

**Comment by Hunter13n** [comment]
im having problems with this, it doesnt add the character, am i doing anything wrong.

**Pull request closed without merging**